plugins/woocommerce/tests/php/src/Admin/ProductBlockEditor/ProductTemplates/ProductTemplateTest.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Automattic\WooCommerce\Tests\Admin\ProductBlockEditor\ProductTemplates;
+
+use Automattic\WooCommerce\Admin\Features\ProductBlockEditor\ProductTemplate;
+
+use WC_Unit_Test_Case;
+
+/**
+ * Tests for the ProductTemplate class.
+ */
+class ProductTemplateTest extends WC_Unit_Test_Case {
+	/**
+	 * Test is_product_type_supported() with a mismatched post type.
+	 */
+	public function test_is_product_type_supported_post_type_mismatch() {
+		$template = new ProductTemplate(
+			array(
+				'id'        => 'test-product-template',
+				'title'     => 'Test Product Template',
+				'post_type' => 'post',
+			)
+		);
+
+		$this->assertFalse( $template->is_product_type_supported( 'simple', 'product' ) );
+	}
+
+	/**
+	 * Test is_product_type_supported() with a mismatched product type.
+	 */
+	public function test_is_product_type_supported_product_type_mismatch() {
+		$template = new ProductTemplate(
+			array(
+				'id'           => 'test-product-template',
+				'title'        => 'Test Product Template',
+				'post_type'    => 'product',
+				'product_data' => array(
+					'type' => 'variable',
+				),
+			)
+		);
+
+		$this->assertFalse( $template->is_product_type_supported( 'simple', 'product' ) );
+	}
+
+	/**
+	 * Test is_product_type_supported() with a matching product type.
+	 */
+	public function test_is_product_type_supported_product_type_match() {
+		$template = new ProductTemplate(
+			array(
+				'id'           => 'test-product-template',
+				'title'        => 'Test Product Template',
+				'post_type'    => 'product',
+				'product_data' => array(
+					'type' => 'simple',
+				),
+			)
+		);
+
+		$this->assertTrue( $template->is_product_type_supported( 'simple', 'product' ) );
+	}
+
+	/**
+	 * Test is_product_type_supported() with a null product type and matching post type.
+	 */
+	public function test_is_product_type_supported_product_type_null_match() {
+		$template = new ProductTemplate(
+			array(
+				'id'           => 'test-product-template',
+				'title'        => 'Test Product Template',
+				'post_type'    => 'product_variation',
+				'product_data' => array(
+					'type' => null,
+				),
+			)
+		);
+
+		$this->assertTrue( $template->is_product_type_supported( null, 'product_variation' ) );
+	}
+
+	/**
+	 * Test is_product_type_supported() with a matching alternate product type.
+	 */
+	public function test_is_product_type_supported_product_type_match_with_alternate_product_data() {
+		$template = new ProductTemplate(
+			array(
+				'id'                      => 'test-product-template',
+				'title'                   => 'Test Product Template',
+				'post_type'               => 'product',
+				'product_data'            => array(
+					'type' => 'simple',
+				),
+				'alternate_product_datas' => array(
```

Comment:
Yeah. I didn't love that name myself, but as I used it, I got more used to it. It likely just isn't a good name though, my gut tells me!

While the arrays in `alternate_product_datas` could contain data other than `type`, we currently only use `type` when attempting to do (fallback) matching (which is what the `type` is used for, other than when the product is initially created).

So, we could go with `alternate_product_types` instead and flatten it to a simple array.

Or, change the `type` in the `product_data` to be `types`, and special case how we handle that, so that we use the first array element as the type when creating a new product with that template and the other array elements are only used as alternates.

All of this is really for the `simple` -> `variable` type change that we do under the hood (since in both cases, we want the user to think of this as a "Standard Product"). Though, there may be other scenarios where this would happen, when extensions are considered. So, whatever we do, we should leave the door open for extensibility needs.